src/master-playlist-controller.js

```diff
@@ -1066,6 +1075,9 @@ export class MasterPlaylistController extends videojs.EventTarget {
 
     // Blacklist this playlist
     currentPlaylist.excludeUntil = Date.now() + (blacklistDuration * 1000);
+    if (error.reason) {
+      currentPlaylist.lastExcludeReason = error.reason;
```

Comment:
For the short term, do you think we should add this as `lastExcludeReason_` to prevent others from using it? Or maybe we should decide if it would make sense to have an array of reasons/what we want a public API on playlists should be.

Reply:
I think making it an underscore works as I think our whole exclusion mechanism needs a rework and I don't want anyone to rely on it.

Reply:
yeah, let's keep it private for now. We should think about this more formally for the longer term.